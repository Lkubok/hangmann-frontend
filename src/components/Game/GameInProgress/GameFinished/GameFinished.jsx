import React, { Component } from "react";
import { connect } from "react-redux";
import * as selectors from "../../../../reducers/selectors";
import "./GameFinished.scss";
import { clearGameParams } from "../../../../actions/gameActions";
import { fetchSingleQuote } from "../../../../actions/gameActions";

const { REACT_APP_API_HOST } = process.env;

export class GameFinished extends Component {
  handlePlayAgain = () => {
    const { userName, userEmail, userLevel, userLang } = this.props;
    this.props.clearGameParams(userName, userEmail, userLevel, userLang);
  };
  searchedQuote = id => {
    const idToFetch = id.slice(0, 24);
    this.props.fetchSingleQuote(REACT_APP_API_HOST, idToFetch);
    return this.props.searchedQuote;
  };
  componentDidMount() {
    //TODO:
    // - SEND HERE STATS FROM GAME TO BACKEND
  }
  render() {
    const { isGuessed, gameId } = this.props;
    return (
      <div
        className={
          isGuessed
            ? "game-finished game-finished-win"
            : "game-finished game-finished-dead"
        }
      >
        <h2>game finished</h2>
        {isGuessed ? <h3>Congratulations !</h3> : <h3>You are dead ;( </h3>}
        {isGuessed ? null : <h4>searched quote was:</h4>}
        {isGuessed ? null : <h5>{this.searchedQuote(gameId)}</h5>}
        <button className="btn-restart" onClick={this.handlePlayAgain}>
          Play again
        </button>
      </div>
    );
  }
}
const mapStateToProps = state => ({
  userName: selectors.getGameUserName(state),
  userEmail: selectors.getUserEmail(state),
  userLevel: selectors.getGameLevel(state),
  userLang: selectors.getGameLang(state),
  isGuessed: selectors.isGuessed(state),
  gameId: selectors.getGameId(state),
  searchedQuote: selectors.getSearchedQuote(state)
});

const mapDispatchToProps = {
  clearGameParams,
  fetchSingleQuote
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(GameFinished);
