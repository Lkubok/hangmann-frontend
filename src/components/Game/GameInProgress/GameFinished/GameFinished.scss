@import "../../../../scss/mixins.scss";
@import "../../../../scss/variables.scss";

.game-finished {
  @include flex-cc-row;
  position: absolute;
  width: 100%;
  min-height: 100%;
  font-weight: 900;
  font-size: 3em;
  align-items: flex-start;
  text-align: center;
  border-radius: 20px;
  &-win {
    background-color: $primary-color-opacity;
  }
  &-dead {
    background-color: $danger-color-opacity;
  }
  h2 {
    text-transform: uppercase;
    padding-top: 20px;
    flex-basis: 100%;
  }
  h3,
  h4,
  h5 {
    flex-basis: 100%;
  }
  h5 {
    color: $secondary-color;
  }
  .btn-restart {
    margin: 40px;
    &:hover {
      background-color: $dark-color-hover;
      color: $white-color;
    }
  }
}
