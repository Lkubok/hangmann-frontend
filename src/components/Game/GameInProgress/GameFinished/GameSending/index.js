import GameSending from "./GameSending";
export default GameSending;
