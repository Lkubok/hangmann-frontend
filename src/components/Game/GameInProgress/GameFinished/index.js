import GameFinished from "./GameFinished";
export default GameFinished;
