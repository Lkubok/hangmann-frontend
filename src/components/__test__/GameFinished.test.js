import React from "react";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { GameFinished } from "../Game/GameInProgress/GameFinished/GameFinished";
Enzyme.configure({ adapter: new Adapter() });

describe("GameFinished component", () => {
  let props;
  const spyFunction = jest.fn();
  beforeEach(() => {
    props = {
      userName: "lucas",
      userEmail: "aa@wp.pl",
      userLevel: "easy",
      userLang: "pl",
      isGuessed: false,
      gameStatus: "win",
      gameId: "123",
      searchedQuote: "test quote",
      scoreToSend: "win",
      startTime: 123,
      isFinished: true,
      fetchSingleQuote: spyFunction
    };
  });
  it("Should match snapshot", () => {
    const wrapper = shallow(<GameFinished {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
