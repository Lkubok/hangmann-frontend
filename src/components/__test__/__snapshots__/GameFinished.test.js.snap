// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameFinished component Should match snapshot 1`] = `
<div
  className="game-finished game-finished-dead"
>
   
  <h2>
    game over !
  </h2>
  <h3>
    You are dead ;( 
  </h3>
  <h4>
    searched quote was:
  </h4>
  <h5>
    test quote
  </h5>
  <button
    className="btn-restart"
    onClick={[Function]}
  >
    Play again
  </button>
  <Connect(GameSending) />
</div>
`;
